### PR TITLE
Refactor testing code

### PR DESCRIPTION
**Pull request by stefsmeets**
Hi all, to familiarize myself a bit with the code I had a look at refactoring the test cases. I reduced the number of cycles, so that the tests run about 4 times faster (<2 min instead of 7 min on my pc). I also cleaned and refactored the code, so that the tests now share most of the setup. And the tests now run in a temporary directory, which avoids cluttering the local directory.

Addresses #31

### CODE REVIEW

**Comment by stefsmeets** [comment]
This PR is ready, @lsoucasse could you have a look and let me know what you think?

**Comment by nichollsh** [comment]
Capturing the runaway plateau is the key behaviour in the runaway test. Physically, this is checking that the temperature profile sits on the saturation curve for H2O, and therefore covers the physics relating to latent heat and the temperature profile integrator. It also tests SOCRATES longwave calculations.

The instellation test captures the surface energy balance stuff relating to conduction through a solid skin layer above the magma ocean. This also tests the SOCRATES longwave and shortwave calculations.

We can probably restrict this to only a few points, as @lsoucasse suggests, as long as they are well placed (i.e. extrema and the plateau in the runaway case, and high/low instellation in the other case).

**Comment by stefsmeets** [comment]
Can you suggest other ways to reduce the runtime for this test, e.g. by reducing the resolution or otherwise? In my opinion, it is still on the long side, especially if there is a need to test multiple starting values.

I'm mostly interested in ensuring that the code gives the expected output. What in the model or code could change that it would give the expected output at the lower extrema, but not at the higher one? Then we should directly test that instead.

**Comment by timlichtenberg** [comment]
The runaway plateau is – unfortunately for this issue – an emergent behaviour that results from a combination of T-P structure, opacity data, and radiative transfer. You need the whole code to achieve these numbers and not a single function can do this. This test is therefore somewhat an integrated test of the overall physical behaviour of the code. Lower resolutions have been previously shown to change the value we aim at here (which is an accepted literature value). So I do not have immediate ideas how to simplify/speed up this test.

**Comment by stefsmeets** [comment]
Okay, it's fairly easy to parametrize the tests now, so I can set it up in a way that new points can be easily added.

It may also be worth having a flag to trigger the full integration test, so it is less disruptive when coding.

**Comment by stefsmeets** [comment]
I have parametrized the tests, so make it easier to add/remove points. I have added the min/max inputs and a middle input to capture the flux plateau. Still slow, but at least a factor 4 improvement over the original code.

Let me know what you think.

File: tests/test_instellation.py [hunk 1]
```diff
@@ -1,96 +1,37 @@
...
-    if os.path.exists(dirs["output"]):
-        shutil.rmtree(dirs["output"])
-    os.mkdir(dirs["output"])
 
-    #Download required spectral files
-    DownloadSpectralFiles("Oak")
-    DownloadStellarSpectra()
+TEST_DATA = (
+    (0.3, (2.34297e03, 1.94397e03, -4.67185e01, 3.00023e03, 4.19568e02)),
+    (1.4, (1.07585e02, 4.15768e02, 3.24365e02, 2.99838e03, 1.94222e02)),
```
Comment (lsoucasse) [inline tests/test_instellation.py:9]:
```suggestion
#    (0.483333, (9.02641e02, 8.43916e02, 7.70272e01, 2.99961e03, 3.30552e02)),
#    (0.666667, (4.74451e02, 5.01731e02, 9.86425e01, 2.99951e03, 2.81454e02)),
#    (0.85, (2.91857e02, 4.21028e02, 1.73069e02, 2.99913e03, 2.49260e02)),
#    (1.03333, (1.97482e02, 4.16319e02, 2.48540e02, 2.99876e03, 2.26070e02)),
#    (1.21667, (1.42451e02, 4.15915e02, 2.94890e02, 2.99853e03, 2.08342e02)),
    (1.4, (1.07585e02, 4.15768e02, 3.24365e02, 2.99838e03, 1.94222e02)),
```

**Review by lsoucasse — APPROVED** [review]

**Pull request merged**